patches/0007-Add-support-for-QMJ5.patch

```diff
@@ -0,0 +1,417 @@
+@@ -184,6 +191,18 @@ public abstract class AbstractRemapJarTask extends Jar {
+ 			outputFile = getParameters().getOutputFile().getAsFile().get().toPath();
+ 		}
+ 
++		protected void convertQmj5ToQmj() throws IOException {
++			ZipUtils.transform(outputFile, Map.of(QUILT_MOD_JSON5_PATH, bytes -> {
++				var jsonReader = new GsonReader(JsonReader.json5(new String(bytes, StandardCharsets.UTF_8)));
++				var element = LoomGradlePlugin.GSON.fromJson(jsonReader, JsonElement.class);
++				var convertedElement = LoomGradlePlugin.GSON.toJson(element);
++
++				return convertedElement.getBytes();
++			}));
++
++			ZipUtils.move(outputFile, QUILT_MOD_JSON5_PATH, QUILT_MOD_JSON_PATH);
++		}
++
+ 		protected void modifyJarManifest() throws IOException {
+ 			int count = ZipUtils.transform(outputFile, Map.of(MANIFEST_PATH, bytes -> {
+ 				var manifest = new Manifest(new ByteArrayInputStream(bytes));
+diff --git a/src/main/java/net/fabricmc/loom/task/RemapJarTask.java b/src/main/java/net/fabricmc/loom/task/RemapJarTask.java
+index 319fde2eda353146b917b2cb59149083b7ea5430..04692b0147c6b579a1693349dad131a9f9eb7557 100644
+--- a/src/main/java/net/fabricmc/loom/task/RemapJarTask.java
++++ b/src/main/java/net/fabricmc/loom/task/RemapJarTask.java
+@@ -215,6 +215,7 @@ public abstract class RemapJarTask extends AbstractRemapJarTask {
+ 					markClientOnlyClasses();
+ 				}
+ 
++				convertQmj5ToQmj();
+ 				remapAccessWidener();
+ 				addRefmaps();
+ 				addNestedJars();
+diff --git a/src/main/java/net/fabricmc/loom/task/RemapSourcesJarTask.java b/src/main/java/net/fabricmc/loom/task/RemapSourcesJarTask.java
+index cca9638ed583a3e25e3142f02188293f2c23a6c5..7138ecdddcff8cd9959408c4205552882f014d2e 100644
+--- a/src/main/java/net/fabricmc/loom/task/RemapSourcesJarTask.java
++++ b/src/main/java/net/fabricmc/loom/task/RemapSourcesJarTask.java
+@@ -87,6 +87,7 @@ public abstract class RemapSourcesJarTask extends AbstractRemapJarTask {
+ 			try {
+ 				sourceRemapperService.remapSourcesJar(inputFile, outputFile);
+ 
++				convertQmj5ToQmj();
+ 				modifyJarManifest();
+ 				rewriteJar();
+ 			} catch (Exception e) {
+diff --git a/src/main/java/net/fabricmc/loom/util/ZipUtils.java b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
+index f6214aed8a99b99af52dcca76d78a36c2cf29213..14e770e5dc42faac1ee3bc91250d280afc89cc9e 100644
+--- a/src/main/java/net/fabricmc/loom/util/ZipUtils.java
++++ b/src/main/java/net/fabricmc/loom/util/ZipUtils.java
+@@ -47,6 +47,8 @@ import org.jetbrains.annotations.Nullable;
+ import org.objectweb.asm.ClassReader;
+ import org.objectweb.asm.ClassVisitor;
+ import org.objectweb.asm.ClassWriter;
++import org.quiltmc.parsers.json.JsonReader;
++import org.quiltmc.parsers.json.gson.GsonReader;
+ 
+ import net.fabricmc.loom.LoomGradlePlugin;
+ 
+@@ -112,6 +114,13 @@ public class ZipUtils {
+ 		return LoomGradlePlugin.GSON.fromJson(new String(bytes, StandardCharsets.UTF_8), clazz);
+ 	}
+ 
++	@SuppressWarnings("unchecked")
```

Comment:
dammit! i forgot to remove these;
i'll remove those once there is another change to be suggested